packages/itwin/property-grid/src/hooks/UseContextMenu.tsx

```diff
@@ -24,27 +23,37 @@ export enum PropertyGridDefaultContextMenuKey {
   ShowNull = "show-null",
 }
 
-export type ContextMenuItemInfo = ContextMenuItemProps & {
-  key?: string | number;
+/** Data structure that defined single context menu item. */
+export interface ContextMenuItemDefinition {
+  key: string;
   label: string;
-  isValid?: (record: PropertyRecord, field?: Field) => boolean;
-  forcePosition?: number;
-};
+  title?: string;
+  execute: () => void;
+}
+
+/** Context provided to menu items for performing actions and determining if item should be shown. */
+export interface MenuItemContext {
+  /** iModelConnection used by property grid. */
+  imodel: IModelConnection;
+  /** Data provider used by property grid. */
+  dataProvider: IPresentationPropertyDataProvider;
+  /** Property record for which menu is opened. */
+  record: PropertyRecord;
+  /** Field for which context menu is opened. */
+  field: Field | undefined;
+  /** Specified whether `null` values are shown in property grid. */
+  showNullValues: boolean;
+  /** Callback that changes whether property grid should show `null` values or not. */
+  setShowNullValues: (values: boolean) => Promise<void>;
```

Comment:
AFAIK, this is required for the "show/hide null properties" context menu buttons, which, IMO, don't belong to context menu at all. Filed [an issue](https://github.com/iTwin/viewer-components-react/issues/499) to solve that separately. In the meantime I believe we can remove these attributes and null value - related context menu item providers altogether.